lib/data/entity.js

```diff
@@ -354,20 +354,32 @@ const ConflictType = {
   SOFT: 'soft', // discrepancy in version number but no overlaping properties
   HARD: 'hard' // discrepancy in version number and one/more properties updated simultaneously
 };
-const getWithConflictDetails = (defs) => {
+const getWithConflictDetails = (defs, audits, relevantToConflict) => {
   const defMap = new Map(defs.map(d => [d.version, d]));
 
   const result = [];
 
+  const lastResolveEvent = audits.findLast(a => a.action === 'entity.update.resolve');
+
+  const auditMap = new Map(audits.map(a => [a.details.entityDefId, a.details]));
+
+  const versionMap = new Map();
```

Comment:
Is `defs` guaranteed to be ordered by version number? If so, instead of creating a separate map, could we look up each version in the array by index using the version's version number? Or are there cases where that might not work?

Reply:
Sort of a different question, but relevant if we decide we need to keep `versionMap`: what's the difference between `versionMap` and `defMap`? Could we just use one or the other?

Reply:
`defMap` values are incoming Entity.Def frames and `versionMap` values are results with all additional fields. 

Regarding `versionMap`, I created it in a fear of missing versions. Now after thinking a bit, I feel it's an unfounded fear and I should use array indexing in both places instead. Can you think of any situation where we would have missing version number?

Reply:
I can't think of a case where a version number would be outright missing. The version number is incremented one by one, and versions can't be deleted, so I don't see how a number could get skipped.

Another issue would be if versions are out-of-order (e.g., somehow 1, 3, 2, not 1, 2, 3). But I again don't see that happening, since I think we sort on `entity_defs."createdAt", entity_defs.id`, and version numbers are incremented one by one. In order to increment the version number of version 2, version 2 has to have been created previously.

In conditions of high concurrency, is there any chance that there could be duplicate version numbers? That would definitely be a problem. What do you think about adding a uniqueness constraint on `entityId` + `version` to the `entity_defs` table? (Could be a separate PR.) Or is it already guaranteed that duplicate version numbers are not possible?

Reply:
Agree, we need to ensure duplicate that there's no duplicate version. I'll create separate PR for that.